contracts/multisig-prover/src/encoding/sol/mod.rs

```diff
@@ -0,0 +1,115 @@
+use std::collections::BTreeMap;
+
+use axelar_encoding::types::{
+    CrossChainId, Message, Payload, PublicKey, Signature, Signer, WeightedSignature, WorkerSet,
+    U256,
+};
+
+use itertools::Itertools;
+use multisig::{msg::SignerWithSig, verifier_set::VerifierSet};
+
+const ECDSA_COMPRESSED_PUBKEY_LEN: usize = 33; // this should be probably a public constant in axelar_encoding.
+const ED25519_PUBKEY_LEN: usize = 32; // this should be probably a public constant in axelar_encoding.
+
+pub fn to_worker_set(vs: &VerifierSet) -> WorkerSet {
+    let signers: BTreeMap<String, Signer> = vs
+        .signers
+        .iter()
+        .map(|(address, signer)| {
+            let enc_signer = signer.address.to_string();
+            let enc_pubkey = to_pub_key(&signer.pub_key);
+
+            let enc_weight = U256::from_be(to_u256_be(signer.weight.u128()));
+
+            (
+                address.clone(),
+                Signer::new(enc_signer, enc_pubkey, enc_weight),
+            )
+        })
+        .collect();
+
+    WorkerSet::new(
+        vs.created_at,
+        signers,
+        U256::from_be(to_u256_be(vs.threshold.u128())),
+    )
+}
+
+fn to_pub_key(pk: &multisig::key::PublicKey) -> PublicKey {
```

Comment:
This file contains all this helpers methods. I did not use `From<T> impl ..` because the being in a third crate it violates the orphan rule.